### PR TITLE
[SPARK-10938] [SQL] remove typeId in columnar cache

### PR DESCRIPTION
**Pull request by davies**
This PR remove the typeId in columnar cache, it's not needed anymore, it also remove DATE and TIMESTAMP (use INT/LONG instead).

### CODE REVIEW

**Comment by davies** [comment]
cc @liancheng @rxin

File: project/MimaExcludes.scala [hunk 1]
```diff
@@ -42,7 +42,9 @@ object MimaExcludes {
         excludePackage("org.spark-project.jetty"),
         MimaBuild.excludeSparkPackage("unused"),
         // SQL execution is considered private.
-        excludePackage("org.apache.spark.sql.execution")
+        excludePackage("org.apache.spark.sql.execution"),
+        // SQL columnar is considered private.
+        excludePackage("org.apache.spark.sql.columnar")
```
Comment (rxin) [inline project/MimaExcludes.scala]: can we move this into execution.columnar?

Comment (davies) [inline project/MimaExcludes.scala]: I'd like not to move it by this reason.

Comment (davies) [inline project/MimaExcludes.scala]: Are you suggesting that we should move everything except public API into execution? That sounds good.

**Comment by cloud-fan** [comment]
LGTM

File: project/MimaExcludes.scala (hunk 1 above)
Comment (rxin) [inline project/MimaExcludes.scala]: Yup. It would be great to minimize the number of top packages we have. You can do it in a followup pr too.

**Pull request closed without merging**